### PR TITLE
Nicer target presentation for Visual Studio

### PR DESCRIPTION
**Pull request by hsorby**
Addresses issue #69

### CODE REVIEW

**Comment by nickerso** [comment]
Looks good to me.

**Comment by jonc125** [comment]
Nice, I'll have to remember this trick!

**Comment by agarny** [comment]
Indeed, it's much better that way:

image

File: cmake/common.cmake [hunk 1]
```diff
@@ -45,3 +45,29 @@ function(HIDE_DISTRACTING_VARIABLES)
...
+        set(_F_PREFIX "Generated Header Files\\")
+      endif()
+      get_filename_component(_G ${_L} DIRECTORY)
+      if(_G)
+        string(REPLACE "/" "\\" _G ${_G})
+      endif()
+      source_group("${_F_PREFIX}${_G}" FILES "${_F}")
+    endforeach()
+  endif()
+endfunction()
```
Comment (agarny) [inline cmake/common.cmake:72]: I would prefer having 'proper' variable names, i.e. replace `_F` with `FILE`, `_F_PREFIX` with `FILE_PREFIX`, etc.

**Comment by agarny** [comment]
Apart from my minor comment above, it's all fine with me.

File: cmake/common.cmake (hunk 1 above)
Comment (hsorby) [inline cmake/common.cmake:72]: I don't think using FILE would be appropriate, CMake might have something to say about that!  I'll change the variable names to something 'proper' but not dangerous.

Comment (agarny) [inline cmake/common.cmake:72]: Actually, CMake is smart enough to know when `FILE` is used as a command or as a variable. This being said, I am happy for you to use something different as long as it is 'proper'. :)

**Comment by hsorby** [comment]
I think this one is ready.

**Comment by mirams** [comment]
closes #69

**Pull request merged**